modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/SchemaDocumenter.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.capability.xml.schema;
+
+import static org.mule.module.extension.internal.capability.xml.schema.AnnotationProcessorUtils.getFieldsAnnotatedWith;
+import static org.mule.module.extension.internal.capability.xml.schema.AnnotationProcessorUtils.getJavaDocSummary;
+import static org.mule.module.extension.internal.capability.xml.schema.AnnotationProcessorUtils.getMethodDocumentation;
+import static org.mule.module.extension.internal.capability.xml.schema.AnnotationProcessorUtils.getMethodsAnnotatedWith;
+import org.mule.api.MuleRuntimeException;
+import org.mule.config.i18n.MessageFactory;
+import org.mule.extension.annotations.Operation;
+import org.mule.extension.annotations.Parameter;
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.declaration.ConfigurationDeclaration;
+import org.mule.extension.introspection.declaration.Declaration;
+import org.mule.extension.introspection.declaration.OperationDeclaration;
+import org.mule.extension.introspection.declaration.ParameterDeclaration;
+
+import java.util.Collection;
+import java.util.Map;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+
+/**
+ * Utility class that picks a {@link Declaration}
+ * on which a {@link Extension} has already been described
+ * and enriches such description with the javadocs extracted from the extension's acting classes.
+ * <p/>
+ * This is necessary because such documentation is not available on runtime, thus this class
+ * uses the annotation processor's AST access to extract it
+ *
+ * @since 3.7.0
+ */
+final class SchemaDocumenter
+{
+
+    private ProcessingEnvironment processingEnv;
+
+    SchemaDocumenter(ProcessingEnvironment processingEnv)
+    {
+        this.processingEnv = processingEnv;
+    }
+
+    void document(Declaration declaration, TypeElement extensionElement)
+    {
+        declaration.setDescription(getJavaDocSummary(processingEnv, extensionElement));
+        documentConfigurations(declaration, extensionElement);
+        documentOperations(declaration, extensionElement);
+    }
+
+    private void documentOperations(Declaration declaration, TypeElement extensionElement)
+    {
+        final Map<String, ExecutableElement> methods = getMethodsAnnotatedWith(extensionElement, Operation.class);
+
+        try
+        {
+            for (OperationDeclaration operation : declaration.getOperations())
+            {
+                ExecutableElement method = methods.get(operation.getName());
+
+                if (method == null)
```

Comment:
There were two cases, one of which I don't remember. The one I remember is the case in which the operation was not added per annotation analysis but because some capability/feature of the API implementation generated it (synthetic operation)